api/python/quilt3/packages.py

```diff
@@ -1428,6 +1433,13 @@ def check_latest_hash():
         top_hash = self._calculate_top_hash(pkg._meta, self.walk())
         pkg._origin = PackageRevInfo(str(registry.base), name, top_hash)
 
+        if dedupe:
+            latest_hash = get_latest_hash()
+            if top_hash == latest_hash:
+                if print_info:
+                    print(f"Package with hash {latest_hash} already exists at the destination; skipping.")
```

Comment:
```suggestion
                    print(f"Skipping since package with hash {latest_hash} already exists at the destination and dedupe is true.")
```